system/datatypes/mailinglist.php

```diff
@@ -702,7 +702,13 @@ protected function mailingView(HyphaRequest $request) {
 			$num = count($addresses);
 
 			$linkToMailing = $this->path(self::PATH_MAILS_VIEW_ID, ['id' => $mailingId]);
-			$action = 'if(confirm(' . json_encode(__('ml-sure-to-send', ['count' => $num])) . '))' . makeAction($linkToMailing, self::CMD_SEND, '');
+
+			$action = 'if(confirm(' . json_encode(__('ml-sure-to-send', ['count' => $num])) . ')){
+				const checkProgress = ()=>{}; // TODO do a call to get the progress and display this in a bar
+				setInterval(checkProgress,1000); //TODO make into a constant
+				window.onbeforeunload = ()=>\'Are you sure you want to leave?\';' // lock the browser from exiting. TODO translation?
```

Comment:
If this is shown to the user, it should probably be more explicit about what happens when you leave?